### PR TITLE
New package version: mfem v4.4

### PR DESCRIPTION
**Pull request by v-dobrev**
Modifications:
- [x] Test CUDA builds
- [x] Test HIP builds
- [x] Test with all external dependencies

### CODE REVIEW

**Comment by eugeneswalker** [comment]
I think this needs a rebase to overcome known broken specs -- this looks to be why the data-viz-sdk CI generation step failed.

**Comment by v-dobrev** [comment]
> I think this needs a rebase to overcome known broken specs -- this looks to be why the data-viz-sdk CI generation step failed.

I merged `develop` into this branch -- that should have the same effect as rebasing.

**Comment by alalazo** [comment]
@v-dobrev Just fyi, since a while you can convert the PR to [draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) instead of marking it WIP

**Comment by v-dobrev** [comment]
This PR is now ready for review. (I removed the `WIP` label.)

Edit: Oops, typo: "is not" --> "is now". One letter completely inverted the meaning. 😄

**Review by tzanio — DISMISSED** [review]
Thank you @v-dobrev

**Review by acfisher — DISMISSED** [review]

**Comment by v-dobrev** [comment]
@tzanio, @acfisher, can you please re-approve -- my last commit (that changes just the tesing script `.../mfem/test_builds.sh`) marked your reviews as stale.

**Review by tzanio — DISMISSED** [review]

File: var/spack/repos/builtin/packages/mfem/package.py [hunk 1]
```diff
@@ -257,29 +297,34 @@ class Mfem(Package, CudaPackage, ROCmPackage):
...
     for gfx in ROCmPackage.amdgpu_targets:
         depends_on('raja+rocm amdgpu_target={0}'.format(gfx),
                    when='+raja+rocm amdgpu_target={0}'.format(gfx))
 
-    depends_on('libceed@0.6:', when='@:4.1+libceed')
-    depends_on('libceed@0.7:', when='@4.2.0:+libceed')
+    depends_on('libceed@0.6', when='@:4.1+libceed')
+    depends_on('libceed@0.7:0.8', when='@4.2.0+libceed')
+    depends_on('libceed@0.8:0.9', when='@4.3.0+libceed')
+    depends_on('libceed@0.10:', when='@4.4.0:+libceed')
```
Comment (jeremylt) [inline var/spack/repos/builtin/packages/mfem/package.py]:
```suggestion
    depends_on('libceed@0.10.1:', when='@4.4.0:+libceed')
```

**Review by acfisher — DISMISSED** [review]

**Comment by v-dobrev** [comment]
@tzanio, @acfisher, as we discussed offline, I updated the conflict with `^mpich@4:` to mfem <= 4.3.

If all the CI checks pass, can you please re-approve one more time?

**Comment by tzanio** [comment]
Thanks @v-dobrev !

**Review by acfisher — APPROVED** [review]

**Pull request merged**